tests/unit/lms/services/course_test.py

```diff
@@ -28,7 +29,7 @@ def test_it_does_nothing_if_theres_already_a_matching_row(
         self, ai_getter, pyramid_request, svc
     ):
         existing_course = factories.Course(
-            consumer_key=pyramid_request.lti_user.oauth_consumer_key,
+            application_instance__consumer_key=pyramid_request.lti_user.oauth_consumer_key,
             authority_provided_id="test_authority_provided_id",
         )
```

Comment:
I don't think we need to use the magic here, we can just instantiate this course like we do the others in the fixture using the `application_instance` fixture which contains the logic presented here. This makes it a simpler.

```suggestion
            application_instance=application_instance,
            authority_provided_id="test_authority_provided_id",
        )
```

I can't put the suggestion in full as github doesn't let you put suggestions over removed lines, but you'd need to pull in the fixture in the args for this to work.